addons/web_editor/static/src/js/backend/convert_inline.js

```diff
@@ -1174,6 +1160,67 @@ function _applyColspan(element, colspan, tableWidth) {
     element.style.setProperty('max-width', width + 'px');
     element.classList.toggle('o_converted_col', true);
 }
+/**
+ * Take an element with a background image and return a string containing the
+ * VML code to display the same image properly in Outlook, with its contents
+ * inside.
+ * TODO: We assume:
+ *      - background-size: cover,
+ *      - background-repeat: no-repeat,
+ *      - size 100%
+ *      - content is centered x/y
+ * These assumptions may not hold for cover snippet and other uses. This needs
+ * to account for that.
+ * TODO: centering span probably not needed with `v-text-anchor:middle` present.
+ *
+ * @param {Element} backgroundImage
+ * @returns {string}
+ */
+function _backgroundImageToVml(backgroundImage) {
+    // Create the outer structure.
+    const clone = backgroundImage.cloneNode(true);
+    const div = document.createElement('div');
+    div.replaceChildren(...clone.childNodes);
+    [['fontSize', 0], ['height', '100%'], ['width', '100%']].forEach(([k, v]) => div.style[k] = v);
+    const vmlContent = document.createElement('div');
+    vmlContent.append(div);
+
+    // Preserve important inherited properties without ancestor context.
+    const style = getComputedStyle(backgroundImage);
+    for (const prop of FONT_PROPERTIES_TO_INHERIT) {
+        div.style[prop] = backgroundImage.style[prop] || style[prop];
+    }
+    [...div.children].forEach(child => child.style.setProperty('font-size', child.style.fontSize || style.fontSize));
+
+    // Prepare the top element for hosting the VML image.
+    for (const prop of ['background', 'background-image', 'background-repeat', 'background-size']) {
+        clone.style.removeProperty(prop);
+    }
+    clone.style.padding = 0;
+    clone.className = clone.className.replace(/p[bt]\d+/g, ''); // Remove padding classes.
+    const url = backgroundImage.style.backgroundImage.match(/url\("?(.+?)"?\)/)[1];
```

Comment:
Are you sure it will always match ? Because if it doesn't then 💥

Reply:
I'd check it to be sure, yes.

Reply:
So that would mean something like this (I didn't indent after the new `if` so you don't have a crazy diff to read):
```diff
diff --git a/addons/web_editor/static/src/js/backend/convert_inline.js b/addons/web_editor/static/src/js/backend/convert_inline.js
index 1d78f06689f..faa9a52b68c 100644
--- a/addons/web_editor/static/src/js/backend/convert_inline.js
+++ b/addons/web_editor/static/src/js/backend/convert_inline.js
@@ -712,13 +712,16 @@ async function toInline($editable, cssRules, $iframe) {
  */
 function flattenBackgroundImages(editable) {
     const backgroundImages = [...editable.querySelectorAll('*[style*=background-image]')]
-        .filter(el => el.style.backgroundImage.includes('url(') && !el.closest('.mso-hide'))
+        .filter(el => !el.closest('.mso-hide'))
         .reverse();
     for (const backgroundImage of backgroundImages) {
-        // Put the Outlook version after the original one in an mso conditional.
-        backgroundImage.after(_createMso(_backgroundImageToVml(backgroundImage)));
-        // Hide the original element for Outlook.
-        backgroundImage.classList.add('mso-hide');
+        const vml = _backgroundImageToVml(backgroundImage);
+        if (vml) {
+            // Put the Outlook version after the original one in an mso conditional.
+            backgroundImage.after(_createMso(_backgroundImageToVml(backgroundImage)));
+            // Hide the original element for Outlook.
+            backgroundImage.classList.add('mso-hide');
+        }
     }
 }
 /**
@@ -1196,7 +1199,9 @@ function _backgroundImageToVml(backgroundImage) {
     }
     clone.style.padding = 0;
     clone.className = clone.className.replace(/p[bt]\d+/g, ''); // Remove padding classes.
-    const url = backgroundImage.style.backgroundImage.match(/url\("?(.+?)"?\)/)[1];
+    const matches = backgroundImage.style.backgroundImage.match(/url\("?(.+?)"?\)/);
+    const url = matches && matches[1];
+    if (url) {
     clone.setAttribute('background', url);
     clone.setAttribute('valign', 'middle');
 
@@ -1218,6 +1223,7 @@ function _backgroundImageToVml(backgroundImage) {
 
     // Wrap the VML in the original opening and closing tags.
     return `${clone.outerHTML.replace(/<\/[\w-]+>[\s\n]*$/, '')}${vml}</${clone.nodeName.toLowerCase()}>`;
+    }
 }
 /**
  * Take a selector and return its specificity according to the w3 specification.
```
Is that ok what you had in mind?